### PR TITLE
Add stats to task API

### PR DESCRIPTION
**Pull request by nickpetrovic**
- Add `stats.queue_depth` to task get and list endpoints

Resolve BE-1487

### CODE REVIEW

**Comment by nickpetrovic** [comment]
Preview of the `stats.tasks_in_flight` field. 

image

File: pkg/api/v1/task.go [hunk 1]
```diff
@@ -121,6 +125,16 @@ func (g *TaskGroup) addOutputsToTask(ctx context.Context, workspaceName string,
 	return nil
 }
 
+func (g *TaskGroup) addStatsToTask(ctx context.Context, workspaceName string, task *types.TaskWithRelated) error {
+	tasksInFlight, err := g.taskRepo.TasksInFlight(ctx, workspaceName, task.Stub.ExternalId)
+	if err != nil {
+		return err
+	}
+
+	task.Stats.InFlight = uint32(tasksInFlight)
```
Comment (jsun-m) [inline pkg/api/v1/task.go:134]: In the V1 dashboard, we called this queue depth. Not sure if you like that more than this

Comment (nickpetrovic) [reply pkg/api/v1/task.go:134]: Yeah good thought. @mernit do you have a preference?

Comment (nickpetrovic) [reply pkg/api/v1/task.go]: Renamed to queue_depth.

**Pull request merged**